packages/utils/telemetry-utils/src/sampledTelemetryHelper.ts

```diff
@@ -57,10 +66,12 @@ interface Measurements {
  *
  * @internal
  */
-export class SampledTelemetryHelper implements IDisposable {
+export class SampledTelemetryHelper<TCustomMetrics extends string> implements IDisposable {
 	disposed: boolean = false;
 
-	private readonly measurementsMap = new Map<string, Measurements>();
+	private readonly measurementsMap = new Map<string, LoggerData<TCustomMetrics>>();
+	private readonly dataSums: { [key in TCustomMetrics]?: number } = {};
+	private readonly dataMaxes: { [key in TCustomMetrics]?: number } = {};
```

Comment:
These need to be kept inside the objects which are measurementMaps' values, because each bucket needs its own accumulators.